src/Lesson6/Animal.java

```diff
@@ -0,0 +1,43 @@
+package Lesson6;
+
+public abstract class Animal {
+  protected final String name;
+  private static int animalsCount = 0;
+  private static int catsCount = 0;
+  private static int dogsCount = 0;
+
+  public Animal(String name) {
+    this.name = name;
+
+    animalsCount++;
+  }
+
+  protected void setCatsCount() {
```

Comment:
```suggestion
  protected void incrementCatsCount() {
```
поскольку мы увеличиваем, а не создаем

Reply:
готово